src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -1161,7 +1159,7 @@ internal static TResult ExecuteScriptAndGetItem<TResult>(string scriptToExecute,
             using (PowerShell pwsh = PowerShell.Create())
             {
                 pwsh.Runspace = runspace;
-                IEnumerable<TResult> results = pwsh.AddScript(scriptToExecute).Invoke<TResult>();
+                IEnumerable<TResult> results = pwsh.AddScript(scriptToExecute, useLocalScope: true).Invoke<TResult>();
```

Comment:
I assume this isn't supposed to be something that amends the user's environment/context? One worry in the back of my head is that we currently rely on dot-sourcing to populate things from the REPL into the state of the PSIC. If we can't dot-source, I'm not sure what we'll do. Possibly the solution is to only dot-source when the language mode permits?

Reply:
Right. That's what I've been talking about in Teams. If ConstrainedLanguage mode won't let us implement a REPL? How will this work?

Reply:
also we sure that's actually a problem?

```powershell
$iss = [initialsessionstate]::CreateDefault2()
$iss.LanguageMode = 'ConstrainedLanguage'
$rs = $ps = $null
try {
    $rs = [runspacefactory]::CreateRunspace($Host, $iss)
    $ps = [powershell]::Create()
    $rs.Open()
    $ps.Runspace = $rs
    $ps.AddScript('$what = "yes"', $false).Invoke()
} finally {
    if ($null -ne $ps) {
        $ps.Dispose()
    }

    if ($null -ne $rs) {
        $rs.Dispose()
    }
}
```

^ works

Reply:
@SeeminglyScience the error is this:

> Cannot dot-source this command because it was defined in a different language mode. To invoke this command without importing its contents, omit the '.' operator. 

Which I think implies that scripts added via AddScript within PSES are treated as `FullLanguage`... But the `Invoke()` can only invoke `ConstrainedLanguage` scripts because of AppLocker.

Reply:
Right I get that, but have we confirmed that `AddScript(script, false)` always does that? I mean PowerShell's `Executor` does it somehow right? And the example above?

I feel like we're missing something.

Reply:
The docs on AddScript are bad, but the docs on this lead me to believe it's true:

https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.commandinvocationintrinsics.invokescript?view=pscore-6.2.0#System_Management_Automation_CommandInvocationIntrinsics_InvokeScript_System_String_System_Boolean_System_Management_Automation_Runspaces_PipelineResultTypes_System_Collections_IList_System_Object___

Reply:
That dot sourcing you called out only applies to debugging Untitled files and the users say the extension just won't start up so it wouldn't be that.

Reply:
To be clear I'm not questioning whether that parameter causes the script to be dot sourced, I'm questioning whether that API always throws that exception.  More to the point, my example is dot sourcing in `ConstrainedLanguage` so it's clear that it doesn't always.  PowerShell itself also dot sources prompts in `ConstrainedLanguage` no?

Reply:
Given the discussion we had today, does it make sense to undo this part of the change?

Reply:
I think it's still useful. If we can run in a new scope we probably should.

Reply:
But if the thing we execute here is side-effectful, then any state it sets will be lost rather than dot-sourced. Do we use this function in a specific way, or do we execute user-provided scripts with it?

Reply:
none of the references of this method have side-effects. It's 3 references:

$Host.Name
$PSVersionTable
$env:PROCESSOR_ARCH...

Reply:
It still makes me uneasy having one `Execute*` that explicitly creates a new scope without really telegraphing that.  It may be fine with all of the current things it's referencing but we shouldn't rely on remembering that fact down the line. Can we just add a `bool createNewLocalScope = false` parameter?

Reply:
Yeah I'm thinking along the same lines; we have a number of methods called `Execute` and only one of them would have an implicitly different scoping behaviour, and that's not guaranteed to fix the root cause of the issue anyway

Reply:
The main point of language mode is to prevent privilege escalation when running scripts on locked down systems (currently only supported on Windows, but may change in future).  This extension is trusted and so needs to distinguish between known trusted scripts, and arbitrary scripts it receives and must be considered untrusted.  Trusted scripts can be run in FL mode and should be run in its own runspace.  Untrusted scripts can be run in the console runspace that is in CL mode.  Is it possible to make this distinction when `Execute`ing scripts?

Reply:
I've set a default parameter to have a value of false and have added useLocalScope to all 3 references.